### DISCUSSION TITLE
How to delete interval after exit screen

### DISCUSSION THREAD

**Discussion by jakubziebin** [Textualize/textual#3219] [Q&A] [answered]
I have class in one screen which set_interval:
```
class MyCustomClass(MyCustomWidget):
    def __init__(self) -> None:
        super().__init__()
        self.__information = MyData()
        self.set_interval(settings.get("node.refresh_rate", 1.5), lambda: self.on_data_refresh())
```

When I leave the screen where I created an instance of this class, can I delete this interval?

**Comment by davep** [comment, marked as answer]
See [the documentation for `set_interval`](https://textual.textualize.io/api/message_pump/#textual.message_pump.MessagePump.set_interval); you'll notice that it returns a [`Timer`](https://textual.textualize.io/api/timer/#textual.timer.Timer). So you could save that when you call `set_interval` and then [`stop`](https://textual.textualize.io/api/timer/#textual.timer.Timer.stop) it just before you leave the screen.
